### ISSUE TITLE
Add icon for amenity=grit_bin

### ISSUE THREAD

**Issue by lakedistrictOSM** [gravitystorm/openstreetmap-carto#3292]
https://wiki.openstreetmap.org/wiki/Tag:amenity=grit_bin
13.6k uses

Very small feature, so only needs rendering from z19.

Rather horrid rendering proposal [from the wiki](https://wiki.openstreetmap.org/w/index.php?title=Proposed_features/grit_bin&oldid=292642#Representation): ![](https://wiki.openstreetmap.org/w/images/4/40/Grid_bin_32.png)

I'll have a go at some design ideas in the coming weeks, unless anyone wants to jump in first :)

**Comment by Tomasz-W** [comment]
13,6k uses is a lot, but:
* how to show not only a bin, but specifically bin filled of grit?
* do we want to have a 'winter-only' feature on map?

**Comment by lakedistrictOSM** [comment]
> - do we want to have a 'winter-only' feature on map?

Yes. It might be 'winter-only' in some countries, but other countries may need them for most of the year. Also, they exist all year round, even if they're not useful all year round. I think that this will be OK for z19+.

**Comment by matkoniecz** [comment]
I am not convinced that it needs rendering at all on general-purpose map.

**Comment by lakedistrictOSM** [comment]
Do you think that this is recognisable as a grit bin? I'll probably make it smaller (or at least narrower) so that it doesn't use up the full 14x14 canvas but I haven't had much time to play around with it yet.

![gritbin03png](https://user-images.githubusercontent.com/20193716/42724597-23a66ca4-876d-11e8-809f-50d9aac507cf.png)

@dktue @JamesKingdom

**Comment by dktue** [comment]
I love it!

**Comment by matkoniecz** [comment]
> why can't we have a zoomlevel 20

Servers, electricity, [volunteers maintaining hardware](https://github.com/openstreetmap/operations) are not unlimited.

**Comment by Tomasz-W** [comment]
As I'm also sick of the argument "doesn't make sense for a general purpose map" and usually I vote for adding new features to default style, in this case I'm against. It tells something, I think ;)

Please look at proposed icon again and think about it in this way: for us is quite clear design because we are in context and we anticipated grit bin shape, but when you using the map it works opposite - you see some shape and you have to guess intuitively (the quicker, the better) what it is. I think it would be hard to guess that is a grit bin for somebody out of context. Beside of it there are more arguments against which I mentioned above.

**Comment by SomeoneElseOSM** [comment]
> why can't we have a zoomlevel 20 ....

You can - if you render your own tiles.  See https://wiki.openstreetmap.org/wiki/User:SomeoneElse/Tileserver_zoom_levels for everything you'd need to do for zoomlevel 20.  What you can't expect is that donated hardware and services will expand indefinitely to support any potential requirement.

> ... and show almost every mapped feature?

Tomasz-W has explained that pretty well above.  There's no point in showing an icon for something if people can't guess what it is, or worse, think it's something else.  That's problem is compounded by the fact that there's no complete, updated-along-with-the-style legend for the standard layer.

**Comment by SomeoneElseOSM** [comment]
> Though I am not sure whatever somebody asked for zoom level 20.

The maps I create of the UK and Ireland support zoom 24, and I've had comments saying "I'm using your maps because I can see more detail at higher zoom levels", so I'm sure the demand is out there.  "What gets rendered on osm.org" is a completely different issue of course.

**Comment by SomeoneElseOSM** [comment]
Here's another possibility for amenity=grit_bin:

https://github.com/SomeoneElseOSM/openstreetmap-carto-AJT/blob/master/symbols/grit_bin.p.16.png

**Comment by meased** [comment]
> for us is quite clear design because we are in context and we anticipated grit bin shape

This point is key.

I am against the legend solution. It is essentially giving up on good design. If an icon is not recognizable by most people on its own and needs a legend to be understood, then by definition it's not a good icon. Don't get me wrong, legends are great for documenting and clarifying, but they should not be relied upon.

**Comment by lakedistrictOSM** [comment]
Remember that the map will give context too - grit bins are usually at the side of the road.

**Comment by dktue** [comment]
JOSM uses this as icon for grit bins -- I like this one, too!
![image](https://user-images.githubusercontent.com/2878623/42744565-c754fc4c-88cd-11e8-9fbf-0052204e45d4.png)

**Comment by HolgerJeromin** [comment]
I have never used an public grit_bin. 
Most of the icons here are kind of garbage bins to me.

**Comment by Tomasz-W** [comment]
@kocio-pl An intuitive icon for this feature seems to be impossible. Also temporary character of it suggest not rendering grit bins. I propose to close this issue.
